### PR TITLE
ACL-32 Form (Form, Input)

### PR DESCRIPTION
**Pull request by ElGrecode**
### Links
https://autoclub.atlassian.net/browse/ACL-8
https://autoclub.atlassian.net/browse/ACL-32

### Description
--**The problem**:  
We need to create a wrapper around our base, presentational Form elements
--**The Fix**
This PR is how we are going to do it.

### How to test
`yarn storybook`
Checkout out Form Stories

### Additional Notes
1. Something is wrong with Numeric Input.
2. Validation library, please ignore for now. This is something that needs to be unit tested and I will create at ticket for it.
3. There are no tests yet for Form wrapper components, but since this ticket is a priority to reference, let's move it into master and then I'll create the FormWrapper tests afterwards

### Explanation of Jest/Unit Testing
- Tests around FormGroup

### CODE REVIEW

**Comment by ElGrecode** [comment]
Please ask any questions if what I'm doing is unclear. It's really important we all know what's going on here for FormField wrapper components.

File: src/lib/components/ButtonGroup/ButtonGroup.test.js [hunk 1]
```diff
@@ -8,24 +8,58 @@ import {getDOMNodeComputedStyle} from "../../../../test/DOM";
...
+      const backgroundStyle = getDOMNodeComputedStyle(ButtonGroupComponent.getDOMNode(), 'width');
+      expect(backgroundStyle).to.equal('100%');
+    });
   });
 
-  it('has base width of 100%', () => {
-    const backgroundStyle = getDOMNodeComputedStyle(ButtonGroupComponent.getDOMNode(), 'width');
-    expect(backgroundStyle).to.equal('100%');
+  describe("base styles", () => {
+    it ('has 48px height', () => {
```
Comment (aaa-magenic-eleazarj) [inline src/lib/components/ButtonGroup/ButtonGroup.test.js]: should be: has base width of 100%

File: src/lib/components/Button/Button.js [hunk 2]
```diff
@@ -71,16 +78,27 @@ function Button({
...
-  onClick
+  onClick,
 }:propTypes){
   return (
     <MUIButton
-      className={cx('Button', classes.root, className)}
-      classes={classes}
+      className={cx(
+        'Button',
+        classes.root,
```
Comment (aaa-magenic-amielm) [inline src/lib/components/Button/Button.js]: Why do we need to add this to className and to classes?

File: stories/Form/Form.stories.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+stories
+  .add('basic form', () => {
+    return (
+      <AAAPrimaryTheme>
+        <div className="u-background--gray">
+          <Paper className="u-padding--50">
+              <Form
+                validate={handleValidate}
+                onSubmit={handleFormValueSubmission}
+                render={({ allFieldsHaveValues, handleSubmit }) => {
```
Comment (aaa-magenic-eleazarj) [inline stories/Form/Form.stories.js]: Question @aaa-magenic-adams , what if not all fields are required? How should we handle that on form component?

Comment (aaa-magenic-adams) [reply stories/Form/Form.stories.js]: The validation object will not have a key of required for a specific field.

File: src/lib/components/Button/Button.js (hunk 2 above)
Comment (aaa-magenic-adams) [reply src/lib/components/Button/Button.js]: 👍 Changing

File: src/lib/components/ButtonGroup/ButtonGroup.test.js (hunk 1 above)
Comment (aaa-magenic-adams) [reply src/lib/components/ButtonGroup/ButtonGroup.test.js]: removed 👍

**Pull request merged**